pages/collective-page.js

```diff
@@ -46,13 +49,31 @@ const GlobalStyles = createGlobalStyle`
  * to render `components/collective-page` with everything needed.
  */
 class CollectivePage extends React.Component {
-  static getInitialProps({ req, res, query: { slug, status, step, mode } }) {
+  static async getInitialProps({ client, req, res, query: { slug, status, step, mode } }) {
     if (res && req && (req.language || req.locale === 'en')) {
       res.set('Cache-Control', 'public, s-maxage=300');
     }
 
     if (req) {
       req.noStyledJsx = true;
+
+      // SSR optimization
+
+      // Preload CollectivePage data already
+      const result = await client.query({
+        query: collectivePageQuery,
+        variables: getCollectivePageQueryVariables(slug),
+      });
+      const collective = result?.data?.Collective;
+
+      // Preload Budget section if needed
+      if (collective && getFilteredSectionsForCollective(collective).includes('budget')) {
+        await client.query({
+          query: budgetSectionQuery,
+          variables: { slug, limit: 3 },
```

Comment:
Would be great to have this `3` as a shared constant to make sure someone changing the number of transactions displayed will not break this improvement